File: Algorithms/Warmup/Plus Minus/plusminus.cpp
#include <iostream>
#include <vector>
#include <iomanip>

using namespace std;
int main() {
    int n;
    cin >> n;

    float pos = 0.0;
    float neg = 0.0;
    float zero = 0.0;
    for (int i = 0; i < n; i++) {
        int t;
        cin >> t;
        if (t == 0) {
            zero++;
        } else if (t > 0) {
            pos++;
        } else {
            neg++;
        }
    }

    vector<float> counts = {pos, neg, zero};
    for (float i : counts) {
        cout << setprecision(6) << i / n << endl;
    }

    return 0;
}

File: Algorithms/Warmup/Plus Minus/plusminus.py
# Given code
n = int(input().strip())
arr = [int(arr_temp) for arr_temp in input().strip().split(' ')]

# Start
pos = 0.0
zero = 0.0
neg = 0.0
for i in arr:
    if i == 0:
        zero += 1
    elif i > 0:
        pos += 1
    else:
        neg += 1

print("%.6f" % (pos / n))
print("%.6f" % (neg / n))
print("%.6f" % (zero / n))

